messaging/src/main/java/org/axonframework/eventhandling/AbstractEventProcessor.java

```diff
@@ -143,6 +146,33 @@ protected final void processInUnitOfWork(List<? extends EventMessage<?>> eventMe
         processInUnitOfWork(eventMessages, unitOfWork, ROOT_SEGMENT);
     }
 
+    protected void processInUnitOfWork(Map<EventMessage<?>, ScopeDescriptor> scopeEvents,
```

Comment:
I'd rename `scopeEvents` to something else, maybe `eventToScopeMap`, `eventToScopeCollection` or something else.